### PR TITLE
fix(child-writer): Reduce env vars passed during elevation

### PR DESCRIPTION
**Pull request by jhermsmeier**
As we're not escaping env vars properly in `permissions.getEnvironmentCommandPrefix()`,
passing the entire environment along can cause another instance of the "stuck at starting"
problem, thus we limit the vars explicitly passed along again, keeping PATH and DEBUG.

Tested on:

- [x] Windows
- [x] Mac OS
- [ ] Linux

Change-Type: patch
Connects To: #1851

### CODE REVIEW

File: lib/child-writer/writer-proxy.js [hunk 1]
```diff
@@ -176,11 +178,6 @@ permissions.isElevated().then((elevated) => {
       ipc.of[process.env.IPC_SERVER_ID].on('connect', () => {
         // Inherit the parent evnironment
         const childEnv = _.assign({}, process.env, {
-
-          // The CLI might call operating system utilities (like `diskutil`),
-          // so we must ensure the `PATH` is inherited.
-          PATH: process.env.PATH,
```
Comment (lurch) [inline lib/child-writer/writer-proxy.js]: Why was this line removed?

File: lib/child-writer/writer-proxy.js [hunk 2]
```diff
@@ -176,11 +178,6 @@ permissions.isElevated().then((elevated) => {
       ipc.of[process.env.IPC_SERVER_ID].on('connect', () => {
         // Inherit the parent evnironment
         const childEnv = _.assign({}, process.env, {
```
Comment (lurch) [inline lib/child-writer/writer-proxy.js]: How come this still uses the "bulk" `process.env`, rather than only copying out specific keys as the change on line 104 does?

Comment (jhermsmeier) [reply lib/child-writer/writer-proxy.js]: Because this is a "normal" child process spawn, whereas the elevation prompt gets the env prepended to the command, which causes the issue as mentioned in the description – hence we can just pump the entire env through here.

File: lib/child-writer/writer-proxy.js (hunk 1 above)
Comment (jhermsmeier) [reply lib/child-writer/writer-proxy.js]: See PR description ;)

File: lib/child-writer/writer-proxy.js (hunk 2 above)
Comment (lurch) [reply lib/child-writer/writer-proxy.js]: Right, read that a couple of times and now I understand ;-)
So we _can_ just pump the entire env through here, but _should_ we? Would it be "better" to only pass in the env-vars that we know we'll need?

File: lib/child-writer/writer-proxy.js (hunk 1 above)
Comment (lurch) [reply lib/child-writer/writer-proxy.js]: The PR description doesn't explain why this line was removed, but your comment above about "just pump the entire env through" does :stuck_out_tongue:
